lib/rules/no-shadow.js

```diff
@@ -109,18 +124,24 @@ module.exports = function(context) {
      * Checks if the given variables are shadowed in the given scope.
      * @param {Array} variables The variables to look for
      * @param {Object} scope The scope to be checked.
-     * @returns {void}
+     * @returns {Array} Variables which are already declared in the given scope.
      */
     function checkShadowsInScope(variables, scope) {
+
+        var shadowVars = [];
```

Comment:
@gcochard I think I just should return passed variables from `checkShadowsInScope` instead of non-passed. I still need list of passed variables in `while` statement to check if any left and interrupt otherwise. Then I can get rid of `diff`.